langchain4j/src/main/java/dev/langchain4j/chain/ConversationalRetrievalChain.java

```diff
@@ -69,10 +73,33 @@ public String execute(String question) {
         return aiMessage.text();
     }
 
-    private static String format(List<TextSegment> relevantSegments) {
+    private static String format(List<TextSegment> relevantSegments, List<String> metadata) {
         return relevantSegments.stream()
-                .map(TextSegment::text)
+                .map(textSegment -> textSegmentToText(metadata, textSegment))
                 .map(segment -> "..." + segment + "...")
                 .collect(joining("\n\n"));
     }
+
+    private static String textSegmentToText(List<String> metadata, TextSegment textSegment) {
+        if (metadata == null) {
+            return textSegment.text();
+        }
+
+        StringBuilder formattedText = new StringBuilder();
+
+        for (String metadataKey : metadata) {
+            String metadataContent = textSegment.metadata(metadataKey);
+            if (metadataContent != null) {
+                formattedText.append(metadataKey).append(": ").append(metadataContent).append("\n");
+            }
+        }
+
+        if (!formattedText.toString().isEmpty()) {
+            formattedText.append("Content: ");
+        }
+
+        formattedText.append(textSegment.text());
+
+        return formattedText.toString();
+    }
```

Comment:
could you please add a few unit tests to make sure this togic works as expected? e.g. without specified metadata keys to include, with 1,2 keys, with 0, 1, 2 segments, etc